charts/spire/charts/spire-server/templates/configmap.yaml

```
@@ -1,126 +1,110 @@
+{{- define "spire-server.yaml-config" -}}
 {{- $root := . }}
-{{- $namespace := include "spire-server.namespace" . }}
-apiVersion: v1
-kind: ConfigMap
-metadata:
-  name: {{ include "spire-server.fullname" . }}
-  namespace: {{ include "spire-server.namespace" . }}
-data:
-  server.conf: |
-    server {
-      bind_address = "0.0.0.0"
-      bind_port = "8081"
-      trust_domain = {{ include "spire-server.trust-domain" . | quote }}
-      data_dir = "/run/spire/data"
-      log_level = {{ .Values.logLevel | quote }}
+server:
+  bind_address: "0.0.0.0"
+  bind_port: "8081"
+  trust_domain: {{ include "spire-server.trust-domain" . | quote }}
+  data_dir: "/run/spire/data"
+  log_level: {{ .Values.logLevel | quote }}
+  jwt_issuer: {{ .Values.jwtIssuer | quote }}
 
-      jwt_issuer = {{ .Values.jwtIssuer | quote }}
+  ca_key_type: {{ .Values.caKeyType | quote }}
+  ca_ttl: {{ .Values.caTTL | quote }}
 
-      ca_key_type = {{ .Values.caKeyType | quote }}
-      ca_ttl = {{ .Values.caTTL | quote }}
+  default_x509_svid_ttl: {{ .Values.defaultX509SvidTTL | quote }}
+  default_jwt_svid_ttl: {{ .Values.defaultJwtSvidTTL | quote }}
 
-      default_x509_svid_ttl = {{ .Values.defaultX509SvidTTL | quote }}
-      default_jwt_svid_ttl = {{ .Values.defaultJwtSvidTTL | quote }}
+  ca_subject:
+    {{- with .Values.ca_subject }}
+    - country: [{{ .country | quote }}]
+      organization: [{{ .organization | quote }}]
+      common_name: {{ .common_name | quote }}
+    {{- end }}
 
-      ca_subject = {
-        {{- with .Values.ca_subject }}
-        country = [{{ .country | quote }}],
-        organization = [{{ .organization | quote }}],
-        common_name = {{ .common_name | quote }},
-        {{- end }}
-      }
+  {{- with .Values.federation }}
+  {{- if eq (.enabled | toString) "true" }}
+  federation:
+    bundle_endpoint:
+      {{ .bundleEndpoint | toYaml | nindent 6 }}
```

Based on the testing pr, it looks like this is the fix for the federation test:

```suggestion
      - {{ .bundleEndpoint | toYaml | nindent 6 }}
```